### DISCUSSION TITLE
Saving network properties to a file and using the information later on.

### DISCUSSION THREAD

**Discussion by Gal-Tzvik** [PMEAL/OpenPNM#2909] [Q&A]
Hello, 
I am trying to save an extracted network to a file (csv, networkx, or any type of file which i can open later on) in order to move on with my workflow from a checkpoint without the need to extract the network all over again. How is this possible, trying to utilize the functions given within the io class do not work for me for some reason. 

Thank you,
Gal

**Comment by jgostick** [comment]
In version 3 of openpnm we moved towards a design that would be much better at this, but it's not quite there yet.  You're best bet is to use the `pickle` library to save the whole workspace.  

To save:

``` python
import pickle as pk
ws = op.Workspace()  # Get handle to workspace
with open('saved_workspace.pnm', 'wb') as f:
    pk.dump(ws, f)
```

To open:

``` python
with open('saved_workspace.pnm', 'rb') as f:
    ws = pk.load(f)
```

**Comment by Gal-Tzvik** [comment]
So i tried it this way, and for simply saving a workspace with a simple project and then uploading this same project this does work. Sadly, i tried extracting a more complex project which includes different phases and algorithms, but then i get an error considering the attributes of the phase. why doesnt the pickling the SettingsAttr associated with the phase?
